Loadtest stableapis

krishna1s opened the pull request:
# Data Plane API - Pull Request
<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [x] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

Commit:
publishing stable api version 2022-11-01 for load test service

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into loadtest_stableapis

Commit:
prettier run

Commit:
spell check

Commit:
changing swagger back to 2.0

krishna1s committed:
adding description

krishna1s committed:
prettier changes

krishna1s committed:
renaming opertation  id

krishna1s committed:
adding x-ms-enum extension

krishna1s committed:
validation fixes

Commit:
renaming examples

Commit:
review changes

Commit:
adding x-ms-parameter-location

Commit:
review fixes

Commit:
removing required field

krishna1s committed:
adding x-ms-enum values

krishna1s committed:
prettier fix

krishna1s committed:
spell fixes

krishna1s committed:
prettier fix

Commit:
adding new metrics and app component apis

Commit:
updating examples

Commit:
removing list resource type api

Commit:
changing to kebab case

Commit:
changing x-ms-parameter-location value for sdk

Commit:
updating example

Commit:
removing resourceId

Commit:
removing resource Id from examples

Commit:
adding missing desc

Commit:
removing read only properties from examples

Commit:
spell fix

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into loadtest_stableapis

Commit:
review fixes

Commit:
mode renaming

Commit:
properties changes in examples

Commit:
renaming

Commit:
Update loadtestservice.json

Commit:
renaming properties

Commit:
renaming property

Commit:
desc changes

Commit:
Merge branch 'main' into loadtest_stableapis

Harshan01 committed:
Add readme SDK changes (#21315)

Co-authored-by: Harshan B <harshanb@microsoft.com>

Commit:
review changes

1. adding file with application octet stream
2.taking fileName as param instead of fileId
3. adding new api for metrics dimension values
4. simplifying metrics api

Commit:
model validation fixes

Commit:
spell fixes

Commit:
updating desc for search and orderby

Commit:
desc changes

Commit:
adding 201 response

Commit:
adding 201 status code

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into loadtest_stableapis

Commit:
sdk consistency changes

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into loadtest_stableapis

Commit:
filename renaming

Commit:
renaming file info

Commit:
adding default pass fail criteria action value

Commit:
changing operation id and adding x-ms-pageable attributes

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into loadtest_stableapis

Commit:
adding Test or Test Run in operation id

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -488,7 +488,7 @@
         ],
         "summary": "Associate an app component (collection of azure resources) to a test",
-        "operationId": "Test_CreateOrUpdateAppComponents",
+        "operationId": "Test_CreateOrUpdateTestAppComponents",
```
in my opinion, as per the current SDKs this should be `Test_CreateOrUpdateTestAppComponent`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -594,7 +594,7 @@
         ],
         "summary": "Configure server metrics for a test",
-        "operationId": "Test_CreateOrUpdateServerMetricsConfig",
+        "operationId": "Test_CreateOrUpdateTestServerMetricsConfig",
```
similar to AppComponents, I think, this should also be `Test_CreateOrUpdateServerMetricsConfig`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -549,7 +549,7 @@
         ],
         "summary": "Get associated app component (collection of azure resources) for the given test.",
-        "operationId": "Test_ListAppComponents",
+        "operationId": "Test_GetTestAppComponents",
```
I think this should be `Test_GetAppComponents`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -655,7 +655,7 @@
         ],
         "summary": "List server metrics configuration for the given test.",
-        "operationId": "Test_ListServerMetricsConfig",
+        "operationId": "Test_GetTestServerMetricsConfig",
```
similarly, I think this should be, `Test_GetTestServerMetricsConfig`

Harshan01 replied:
Should we keep it `Test_GetServerMetricsConfig`? Since client name already has Test or TestRun

krishna1s replied:
changed to LoadTestAdministration_CreateOrUpdateServerMetricsConfig

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -1543,7 +1549,7 @@
         ],
         "summary": "List server metrics configuration for the given test run.",
-        "operationId": "TestRun_ListServerMetricsConfig",
+        "operationId": "TestRun_GetTestRunServerMetricsConfig",
```
`TestRun_GetServerMetricsConfig`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -1482,7 +1488,7 @@
         ],
         "summary": "Configure server metrics for a test run",
-        "operationId": "TestRun_CreateOrUpdateServerMetricsConfig",
+        "operationId": "TestRun_CreateOrUpdateTestRunServerMetricsConfig",
```
`TestRun_CreateOrUpdateServerMetricsConfig`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -1437,7 +1443,7 @@
         ],
         "summary": "Get associated app component (collection of azure resources) for the given test run.",
-        "operationId": "TestRun_ListAppComponents",
+        "operationId": "TestRun_GetTestRunAppComponents",
```
`TestRun_GetAppComponents`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -1376,7 +1382,7 @@
         ],
         "summary": "Associate an app component (collection of azure resources) to a test run",
-        "operationId": "TestRun_CreateOrUpdateAppComponents",
+        "operationId": "TestRun_CreateOrUpdateTestRunAppComponents",
```
`TestRun_CreateOrUpdateAppComponents`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -1263,7 +1266,7 @@
         ],
         "summary": "List the dimension values for the given metric dimension name.",
-        "operationId": "TestRun_ListMetricDimensionValues",
+        "operationId": "TestRun_ListTestRunMetricDimensionValues",
```
`TestRun_ListMetricDimensionValues`

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -1088,7 +1088,7 @@
         ],
         "summary": "List the metric definitions for a load test run.",
-        "operationId": "TestRun_ListMetricDefinitions",
+        "operationId": "TestRun_ListTestRunMetricDefinitions",
```
`TestRun_ListMetricDefinitions`

Commit:
updating operation id

Commit:
adding test suffix in operation id

Commit:
removing the test suffix

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -695,52 +695,12 @@
         ],
         "summary": "Create and start a new test run with the given name.",
-        "operationId": "TestRun_CreateOrUpdate",
+        "operationId": "LoadTestRun_CreateOrUpdate",
```
Should it be, LoadTestRun_CreateOrUpdateTestRun?

krishna1s replied:
changed

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -807,7 +767,7 @@
         ],
         "summary": "Get test run details by name.",
-        "operationId": "TestRun_Get",
+        "operationId": "LoadTestRun_Get",
```
LoadTestRun_GetTestRun?

krishna1s replied:
chnaged

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -844,6 +804,46 @@
+          "TestRun"
+        ],
+        "summary": "Delete a test run by its name.",
+        "operationId": "LoadTestRun_Delete",
```
LoadTestRun_DeleteTestRun?

krishna1s replied:
changed

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -852,7 +852,7 @@
         ],
         "summary": "Get test run file by file name.",
-        "operationId": "TestRun_GetFile",
+        "operationId": "LoadTestRun_GetFile",
```
LoadTestRun_GetTestRunFile?

krishna1s replied:
changed

NiveditJain commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -904,7 +904,7 @@
         ],
         "summary": "Get all test runs with given filters",
-        "operationId": "TestRun_List",
+        "operationId": "LoadTestRun_List",
```
LoadTestRun_ListTestRun?

krishna1s replied:
changed to LoadTestRun_ListTestRuns

NiveditJain reviewed (commented):
As per the current SDK, I have commented changes what I feel, let me know if have decided to move to new names as per the Swagger. Thanks! 

This review closes all my previous comments also.

Commit:
changing operation id

Commit:
changing example response

prativen commented on specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json:
```diff
@@ -852,7 +852,7 @@
         ],
         "summary": "Get test run file by file name.",
-        "operationId": "TestRun_GetFile",
+        "operationId": "LoadTestRun_GetFile",
```
Should this be LoadTestRun_GetTestFile to ensure the right side is symmetric to that of LoadTestAdministration_GetTestFile

krishna1s replied:
changed to LoadTestRun_GetTestRunFile as it is for TestRun se

Commit:
updating operation ids

krishna1s commented:
@mikekistler we have updated the operation to the final version as per our last discussion. Also to keep consistency with LoadTestRun operations we have added the suffix TestRun to all the operations for it so that SDK functions look consistent as below

TestRun and future adding of method

loadTestRun client4 = new LoadTestRun();
client4.CreateOrUpdateTestRun();
client4.CreateOrUpdateAppComponent();

If looks all good, can we merge this also

mikekistler reviewed (approved):
Looks good. 👍

mikekistler commented:
Marked as Approved-BreakingChange as this api-version has not yet been made available to customers.

krishna1s commented:
@tjprescott can you merge this PR

tjprescott commented:
@krishna I can merge this when the PR is green.

krishna1s commented:
/azp run

krishna1s commented:
@tjprescott  should be fine now
